File: Core/AATM.Core.Logging/ILogger.vb
''' <summary>
''' Defines methods for logging messages and exceptions.
''' </summary>
Public Interface ILogger

    ''' <summary>
    ''' Logs a message with informational severity.
    ''' </summary>
    Sub LogInfo(message As String)

    ''' <summary>
    ''' Logs a message with a warning severity.
    ''' </summary>
    Sub LogWarning(message As String)

    ''' <summary>
    ''' Logs an error message.
    ''' </summary>
    Sub LogError(message As String)

    ''' <summary>
    ''' Logs an exception, including its full details.
    ''' </summary>
    Sub LogException(ex As Exception)

End Interface

File: WPF/AATM.Contracts/ILogger.cs
using System;

namespace AATM.Contracts
{
    /// <summary>
/// Defines methods for logging messages and exceptions.
/// </summary>
    public interface ILogger
    {

        /// <summary>
    /// Logs a message with informational severity.
    /// </summary>
        void LogInfo(string message);

        /// <summary>
    /// Logs a message with a warning severity.
    /// </summary>
        void LogWarning(string message);

        /// <summary>
    /// Logs an error message.
    /// </summary>
        void LogError(string message);

        /// <summary>
    /// Logs an exception, including its full details.
    /// </summary>
        void LogException(Exception ex);

    }
}